back/sssscs/src/main/java/com/ib/verification/VerificationCodeService.java

```diff
@@ -81,26 +74,27 @@ private String getCodeText(String code) {
 	@Override
 	public void verifyUser(@Valid VerificationCodeVerifyDTO dto) {
 		User user = userService.findByEmail(dto.getUserEmail());
-		VerificationCode code = get(user);
-		
-		if (isExpired(code)) {
-			repo.delete(code);
-			code = get(user);
-		}
-		
-		if (!code.getCode().equals(dto.getCode())) {
-			decrementAttemptsLeft(user, code);
-			throw new InvalidCodeException();
-		}
+		VerificationCode code = get(user, Reason.TWO_FA, dto.getCode());
 
 		userService.verify(user);
 		userService.resetLoginCounter(user);
 		repo.delete(code);
 	}
 
 	@Override
-	public VerificationCode get(User user) {
-		return repo.findByUser(user).orElseThrow(() -> new VerificationCodeNotFoundException(user));
+	public VerificationCode get(User user, Reason reason, String codeStr) {
+		VerificationCode code = repo.findByUserAndReason(user, reason).orElseThrow(() -> new VerificationCodeNotFoundException(user));
+		if (isExpired(code)) {
+			repo.delete(code);
+			code = get(user, reason, codeStr);
+		}
+		if (codeStr != null && !code.getCode().equals(codeStr)) {
+			if (reason == Reason.TWO_FA) {
```

Comment:
What if the attacker tries to change someone else's password by brute forcing the verification code?

Reply:
https://github.com/magley/sssscs/pull/19/commits/ce74e2c4f4761bc0cee263f599ccf6084cad3d68